### DISCUSSION TITLE
v0.10.0 compatibility

### DISCUSSION THREAD

**Discussion by savchenko** [linrongbin16/fzfx.nvim#701] [General]
Given [this](https://github.com/neovim/neovim/blob/3d39ea3ea9b6e66640e59731d155d731218e7e62/runtime/doc/deprecated.txt#L24-L28)

```vim
DEPRECATED IN 0.10                  *deprecated-0.10*


API
• *nvim_buf_get_option()*   Use |nvim_get_option_value()| instead.
• *nvim_buf_set_option()*   Use |nvim_set_option_value()| instead.
```

would it be safe to assume that snippets like the one below need to be updated?

https://github.com/linrongbin16/fzfx.nvim/blob/2e8991e0fe25bab8f19cc7a6933751719606f8fe/lua/fzfx/commons/api.lua#L18-L24

**Comment by linrongbin16** [comment]
Recently, I also think about this thing.

Since Neovim is deprecating these APIs in its core layer (implemented in C), it looks we will have to keep up with it.

I think I will only support the last legacy major version, while those earlier versions will be dropped.

For example, at the time of writing this comment, 2024-06-04, the stable version is v0.10, nightly is v0.11, the last legacy version is v0.9. So v0.9 is still supported, but earlier versions such as v0.8, v0.7 support will be dropped.

**Reply by linrongbin16** [reply]
Submit an issue: https://github.com/linrongbin16/fzfx.nvim/issues/704

**Reply by linrongbin16** [reply]
Resolved in #706 .
